test/e2e/framework.go

```diff
@@ -622,11 +631,22 @@ func (data *TestData) createPodOnNode(name string, nodeName string, image string
 	// image could be a fully qualified URI which can't be used as container name and label value,
 	// extract the image name from it.
 	imageName := getImageName(image)
+	imageWithRegistry := image
+	if len(testOptions.registry) != 0 {
+		registry := testOptions.registry
+		if strings.Contains(image, "antrea") {
+			imageWithRegistry = fmt.Sprintf("%s/%s", registry, image)
+		} else if image == nginxImage {
+			imageWithRegistry = fmt.Sprintf("%s/antrea/%s", registry, image)
+		} else if image != agnhostImage {
+			imageWithRegistry = fmt.Sprintf("%s/library/%s", registry, image)
+		}
+	}
```

Comment:
This is a bit messy IMO. I think as a first step towards https://github.com/vmware-tanzu/antrea/issues/1555, we should just consider using the images from the VMware registry directly here, instead of introducing a `-registry` option (except if the image is available on `gcr.io`). What do you think?

Reply:
I think adding an option is like what we do in `test-vmc.sh` so it makes it eaiser to choose from Harbor and dockerhub.
Do you think it is messy because there are many `if` here and it is better to hard-code image name with Harbor address? I think we can keep the option and have a map (imageName:imageNameWithHarbor). Then code should be like this:
```
var imagesRegistryMap = map[string]string{
	"nginx":"projects.harbor.vmware.com/antrea/nginx",
}

imageWithRegistry := image
if len(testOptions.registry) != 0 {
    imageWithRegistry = imagesRegistryMap[image]
}
```

Reply:
The thing is e2e tests are also run locally by Antrea developers, and we should make it as easy as possible for them. Things are also a bit more complex for e2e tests because we don't want to use the Harbor registry for all images (e.g. we use an image from `gcr.io`). And even when we use Harbor, some images come from the `antrea` project while others come from the `Library`. So the `-registry` option is not very meaningful IMO. Since we want to try and address #1555 by the 0.12 release, I think it would make sense to start making changes progressively and switch the e2e test images to a functional registry by default. Unless you are concerned about SLA issues with the Harbor registry? It seems that e2e test images are a good place to start this transition (as opposed to updating the Antrea manifests directly). I don't think that sticking with dockerhub as the default makes sense. I also don't think that users / CI need the flexibility to switch registries when it comes to e2e test images: we just need to have *one* option that works for everyone all the time. dockerhub was that one option until they implemented rate limiting, now that one option has to be the VMwarte Distribution Harbor.

Reply:
Thank you for the detailed explanation! You are right. We should make it easier for users to run e2e tests and make Harbor default. Or users will still get the rate limting alert.
Code updated.